### PR TITLE
 [166240824] Enable Social Login via Google and Facebook

### PR DESCRIPTION
**Pull request by Ramadhan0**
##### Have this point working 
* Get ```/api/auth/facebook```: this is the endpoint, a user will hit whenever he/she wants to log in or signup using facebook.
* Get ```/api/auth/google```: this is the endpoint, a user will hit whenever he/she wants to log in or signup using google.
* Get ```/api/auth/twitter```: this is the endpoint, a user will hit whenever he/she wants to log in or signup using twitter.

#### How should this be manually tested?
* Clone the repo by typing  ```git clone https://github.com/andela/savage_rangers-ah``` into the terminal
* Run the server with this command ```npm run dev```
* Test it with your browser by typing:
 
   ```localhost:3001/api/users/google```
   ```localhost:3001/api/users/facebook```
   ```localhost:3001/api/users/twitter```

#### What are the relevant pivotal tracker stories?
[166240824](https://www.pivotaltracker.com/story/show/166240824)

* Test social login in your browser by typing:
### ```localhost:3001/api/users/google```
Screen Shot 2019-06-20 at 08 29 39

### ```localhost:3001/api/users/facebook```
Screen Shot 2019-06-20 at 08 30 28

### ```localhost:3001/api/users/twitter```
Screen Shot 2019-06-20 at 08 31 08

### CODE REVIEW

**Review by truestbyheart — CHANGES_REQUESTED** [review]
Please squash the commits

File: .env.example [hunk 1]
```diff
@@ -4,4 +4,21 @@ DEV_DB_URL=TheDevDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/de
 TEST_DB_URL=TheTestDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/test_dbname
 DB_URL=TheStaggingDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/dbname
 DB_URL=TheProductionDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/dbname
-TOKEN_KEY=TheTokenKey
```
Comment (alainburindi) [inline .env.example]: why did you delete the TOKEN_KEY ??

File: .env.example [hunk 2]
```diff
@@ -4,4 +4,21 @@ DEV_DB_URL=TheDevDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/de
...
+GclientId=google__clientId
+GclientSecret=google__clientSecret
+Gcallback=google__callbackURL
+
+
+TclientId=twitter__consumerKey
+TclientSecret=twitter__consumerSecret
+Tcallback=twitter__callbackURL
+
+secretOrKey=any
```
Comment (alainburindi) [inline .env.example]: is eslint working?? Newline required at end of file but not found

File: src/api/controllers/socialAuth.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if (!findIfExist) {
+      findIfExist = User.create({
+        username: req.user.displayName,
+        email: req.user.emails[0].value,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+    } else {
+      const token = 'generateToken(findIfExist, process.env.TOKEN_KEY)';
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js]: you are putting much useless information into the token. eg: pofileImage

File: src/middlewares/register.app.js [hunk 4]
```diff
@@ -18,6 +18,8 @@ export default (app) => {
     .use(bodyParser.json())
     .use(bodyParser.urlencoded({ extended: true }))
     .use(cors()) // Allow cross origin requests
-    .use(logger('dev')); // Logging http requests
+    .use(logger('dev')) // Logging http requests
+    .use(logger('dev'))
```
Comment (alainburindi) [inline src/middlewares/register.app.js]: why did you repeat .use(logger('dev'))

File: src/middlewares/register.app.js [hunk 5]
```diff
@@ -18,6 +18,8 @@ export default (app) => {
...
+    .set('view engine', 'ejs');
```
Comment (alainburindi) [inline src/middlewares/register.app.js]: we are not yet dealing with views! I need a why

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: test/socialLogin.test.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    models.User.truncate({ cascade: true });
+    models.Token.truncate({ cascade: true });
+    chai
+      .request(server)
+      .post('/testfacebook')
+      .send({
+        id: '1579056305559555'
+      })
+      .end((err, res) => {
+        res.should.have.status(200);
```
Comment (alainburindi) [inline test/socialLogin.test.js]: expect some valid information

**Review by alainburindi — CHANGES_REQUESTED** [review]
work on changes and let me know

File: .travis.yml [hunk 7]
```diff
@@ -1,17 +1,24 @@
...
-  - "10.15.3"
+  - 'stable'
 cache:
   directories:
-    - "node_modules"
+    - 'node_modules'
+env:
+  global:
+    - CC_TEST_REPORTER_ID=67adc75d217c19d11953e75dc6de8182a7ee803a55d94919192c3022b6741ee1
+before_script:
```
Comment (premlogic) [inline .travis.yml]: We've removed the ```before script``` and the ```after sccript```. Remove it too please. Otherwise, you're gonna bring back the Travis errors

File: README.md [hunk 8]
```diff
@@ -2,8 +2,8 @@
 [![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com)
 [![Coverage Status](https://coveralls.io/repos/github/andela/savage_rangers-ah/badge.svg?branch=develop)](https://coveralls.io/github/andela/savage_rangers-ah?branch=develop)
 [![Maintainability](https://api.codeclimate.com/v1/badges/c05960d83e2ec9822111/maintainability)](https://codeclimate.com/github/andela/savage_rangers-ah/maintainability)
-=======
 
+[![Test Coverage](https://api.codeclimate.com/v1/badges/c05960d83e2ec9822111/test_coverage)](https://codeclimate.com/github/andela/savage_rangers-ah/test_coverage)
```
Comment (premlogic) [inline README.md]: This one also. 
PLease, make sure you rebase before you make a push. Because I can see that this line here is not on develop currently.

File: src/api/controllers/socialAuth.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
+/* eslint-disable no-else-return */
+import model from '../models';
+import generateToken from '../../helpers/tokens/generate.token';
+
+const { User } = model;
+
+/**
+ *
```
Comment (premlogic) [inline src/api/controllers/socialAuth.js:8]: Always add a description of your classes/functions @Ramadhan0

File: src/api/controllers/socialAuth.js [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+ *
+ * @class socialLogin
+ */
+class socialLogin {
+  /**
+   * signup via facebook
```
Comment (premlogic) [inline src/api/controllers/socialAuth.js]: A controller to signup a user via facebook

File: src/api/controllers/socialAuth.js [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+   * login  via facebook
```
Comment (premlogic) [inline src/api/controllers/socialAuth.js]: Is this controller for login? or for signup

File: src/api/controllers/socialAuth.js [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+      return res.status(201).json({ user: { ...findIfExist.get(), token } });
+    } else {
+      const loginUser = { username: req.user.displayName, email: req.user.emails[0].value };
+      const token = generateToken(loginUser, process.env.TOKEN_KEY);
+      return res.status(201).json({ user: { ...findIfExist.get(), token } });
+    }
+  }
+
+  /**
+   * signup user via Twitter
```
Comment (premlogic) [inline src/api/controllers/socialAuth.js]: Try to explain more what you are doing in your controller @Ramadhan0

File: src/api/models/user.js [hunk 13]
```diff
@@ -2,12 +2,27 @@ const user = (sequelize, DataTypes) => {
   const User = sequelize.define('Users', {
     username: {
```
Comment (premlogic) [inline src/api/models/user.js]: The username should be unique @Ramadhan0

File: src/api/models/user.js [hunk 14]
```diff
@@ -2,12 +2,27 @@ const user = (sequelize, DataTypes) => {
...
       type: DataTypes.STRING,
+      allowNull: true
     },
     email: {
```
Comment (premlogic) [inline src/api/models/user.js]: Same for the email @Ramadhan0

File: src/api/routes/passport/facebook.js [hunk 15]
```diff
@@ -0,0 +1,13 @@
+import express from 'express';
+import { passport } from '../../../configs/environments';
+import socialLogin from '../../controllers/socialAuth';
+
+// Declaring the app
```
Comment (premlogic) [inline src/api/routes/passport/facebook.js]: Declaring the router.

File: src/api/routes/passport/google.js [hunk 16]
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import { passport } from '../../../configs/environments';
+import socialLogin from '../../controllers/socialAuth';
+// Declaring the app
```
Comment (premlogic) [inline src/api/routes/passport/google.js]: Declaring the router

File: src/api/routes/passport/twitter.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+import express from 'express';
+import { passport } from '../../../configs/environments';
+import socialLogin from '../../controllers/socialAuth';
+// Declaring the app
```
Comment (premlogic) [inline src/api/routes/passport/twitter.js]: Declaring the router

File: src/api/controllers/socialAuth.js [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+    });
+    if (!findIfExist) {
+      findIfExist = await User.create({
+        username: req.user.name.familyName,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+    }
+    const registerUser = { username: req.user.displayName, uniqueId: req.user.id };
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js]: we need id not uniqeId as property

File: src/api/controllers/socialAuth.js [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+    });
+    if (!findIfExist) {
+      findIfExist = User.create({
+        username: req.user.displayName,
+        email: req.user.emails[0].value,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+      const registerUser = { username: req.user.displayName, email: req.user.emails[0].value };
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js]: we need id in the token

File: src/api/controllers/socialAuth.js [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+      }
+    });
+    if (!findIfExist) {
+      findIfExist = User.create({
+        username: req.user.username,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+      const registerUser = { username: req.user.displayName, uniqueId: req.user.id };
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js]: change uniqueId to id

File: src/index.js [hunk 21]
```diff
@@ -1,22 +1,40 @@
...
+import socialAuth from './api/controllers/socialAuth';
+import social from './middlewares/social/social';
 
 const app = express();
-const env = environnements.currentEnv;
-
+const env = currentEnv;
+dotenv.config();
+// session
+app.use(session({
```
Comment (premlogic) [inline src/index.js]: This line should be in ```src/middlewares/register.js```

File: src/index.js [hunk 22]
```diff
@@ -1,22 +1,40 @@
...
+  saveUninitialized: true,
+  secret: process.env.secretOrKey
+}));
 // Register middleware
 register(app);
-
+app.use(passport.initialize());
```
Comment (premlogic) [inline src/index.js]: This one also

File: src/index.js [hunk 23]
```diff
@@ -1,22 +1,40 @@
...
 app.use('/api/', apiRouter);
 app.use('/docs', docsRouter);
 
 app.use('/', homeRouter);
+app.use('/api/auth/', facebookRoutes, googleRoutes, twitterRoutes);
```
Comment (premlogic) [inline src/index.js]: These routers should be in the ```src/api/routes/authRouter``` file

File: src/index.js [hunk 24]
```diff
@@ -1,22 +1,40 @@
...
+
+app.post('/testfacebook', social, socialAuth.socialFacebookAuth);
```
Comment (premlogic) [inline src/index.js]: These routers should be in the routes folder

**Review by alainburindi — CHANGES_REQUESTED** [review]

File: src/middlewares/social/social.js [hunk 25]
```diff
@@ -0,0 +1,35 @@
+export default (req, res, next) => {
```
Comment (premlogic) [inline src/middlewares/social/social.js:7]: @Ramadhan0  please add documentation of this function. But also, I don't think the name of this file describes well what is happening inside

File: test/auth.test.js [hunk 26]
```diff
@@ -75,15 +77,16 @@ describe('Login', () => {
...
       .post('/api/auth/login')
       .send({
         email: 'alain666326@gmail.com',
-        password: 'password',
+        password: 'password'
       })
       .end((err, res) => {
         res.should.have.status(404);
-        res.body.should.have.property('message', 'user doesn\'t exist');
+        res.body.should.have.property('message', "user doesn't exist");
```
Comment (premlogic) [inline test/auth.test.js]: @alainburindi  have to modify this message I think. It doesn't make sense from a user point of view

File: test/home.test.js [hunk 27]
```diff
@@ -23,7 +23,7 @@ describe('Home', () => {
 
 describe('environment', () => {
```
Comment (premlogic) [inline test/home.test.js]: I think if you ignored the env file, you don't need this test

**Review by nkpremices — CHANGES_REQUESTED** [review]

File: .env.example [hunk 28]
```diff
@@ -5,3 +5,20 @@ TEST_DB_URL=TheTestDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/
 DB_URL=TheStaggingDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/dbname
 DB_URL=TheProductionDatabaseConnectionUrl__Format:postgres://user:pass@host:5432/dbname
 TOKEN_KEY=TheTokenKey
+
+
+
+FclientId=facebook__clientId
```
Comment (MemunaHaruna) [inline .env.example]: @Ramadhan0 let's be explicit with the keys. Instead of `FclientId` or `TclientId`, let's go with `FacebookClientId` and `TwitterClientId` etc.

File: .eslintrc.json [hunk 29]
```diff
@@ -1,33 +1,40 @@
 {
```
Comment (MemunaHaruna) [inline .eslintrc.json]: @Ramadhan0 why do you have this file as part of your PR?

File: .travis.yml [hunk 30]
```diff
@@ -1,17 +1,24 @@
 language: node_js
 node_js:
-  - "10.15.3"
+  - 'stable'
```
Comment (MemunaHaruna) [inline .travis.yml]: @Ramadhan0 please, rebase your work with develop.

File: src/api/controllers/socialAuth.js (hunk 18 above)
Comment (MemunaHaruna) [inline src/api/controllers/socialAuth.js]: @Ramadhan0 `registerUser` isn't a good name for this variable. Favour nouns for variable names and verbs for method/function names.

File: src/api/controllers/socialAuth.js [hunk 31]
```diff
@@ -0,0 +1,110 @@
...
+   * signup user via Twitter
+   * login user via Twitter
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - request response
+   * @memberof socialLogin
+   * @returns {object} - the response body
+   */
+  static async socialtwitterAuth(req, res) {
```
Comment (MemunaHaruna) [inline src/api/controllers/socialAuth.js]: *Question:* do we currently have a uniqueId field in the user table? If we don't, you'll have to create a migration to add it.

*Feedback:* 
- The three social login methods have similarities. Can you abstract out the common parts into a separate module to keep the code DRY?

- `socialtwitterAuth` can simply be `twitterAuth` or `twitterLogin`.

- `findIfExist` isn't a good variable name. `existingUser` is better here.

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 29 above)
Comment (MemunaHaruna) [reply .eslintrc.json]: @Ramadhan0 you haven't answered the question.

File: .env.example [hunk 32]
```diff
@@ -11,3 +11,9 @@ MAILER_API_KEY=theTokenOfTheMailerService
 MAILER_EMAIL=theEmailUsedByTheMailerAsSender
 API_BASE_URL=theApiBaseUrlToInsertInTheLinksSharedInEmails
 TOKEN_KEY=TheTokenKey
+
+
```
Comment (MemunaHaruna) [inline .env.example]: @Ramadhan0 I'm confused by the format of these env variables. Why is there a space in front of them? Please, use the format of the other variables in this file and ensure the key names are descriptive.

File: src/api/controllers/db.test.js [hunk 33]
```diff
@@ -0,0 +1,24 @@
+import models from '../models/index';
+
```
Comment (MemunaHaruna) [inline src/api/controllers/db.test.js]: @Ramadhan0 I thought this file was removed in another PR

File: src/api/controllers/password.reset.js [hunk 34]
```diff
@@ -55,19 +55,17 @@ export default class PasswordReset {
     const result = {};
     const userEmail = req.params.email;
     const userPassword = req.body.password;
-    const salt = genSaltSync(parseFloat(process.env.BCRYPT_HASH_ROUNDS) || 10);
+    const salt = genSaltSync(parseFloat(process.env.BCRYPT_HASH_ROUNDS));
```
Comment (MemunaHaruna) [inline src/api/controllers/password.reset.js]: @Ramadhan0 why do you have changes in this file too?

File: src/api/controllers/socialAuth.js [hunk 35]
```diff
@@ -0,0 +1,113 @@
...
+  static async facebookAuth(req, res) {
+    let existingUser = await User.findOne({
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!existingUser) {
+      existingUser = await User.create({
+        username: req.user.name.familyName,
+        profileImage: req.user.photos[0].value,
```
Comment (MemunaHaruna) [inline src/api/controllers/socialAuth.js:32]: @Ramadhan0 what is the format of this `req.user.photos[0].value`? Is it a URL linking to their profile photo? Please, sync with @alainburindi to know his implementation of profile pictures.

File: src/api/controllers/socialAuth.js [hunk 36]
```diff
@@ -0,0 +1,113 @@
...
+      where: {
+        uniqueId: req.user.id
+      }
+    });
+    if (!existingUser) {
+      existingUser = User.create({
+        username: req.user.displayName,
+        email: req.user.emails[0].value,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
```
Comment (MemunaHaruna) [inline src/api/controllers/socialAuth.js:66]: @Ramadhan0 why do we need to store the `provider`?

File: src/api/controllers/socialAuth.js [hunk 37]
```diff
@@ -0,0 +1,113 @@
...
+        uniqueId: req.user.id
+      });
+      const tempUser = { username: req.user.displayName };
```
Comment (MemunaHaruna) [inline src/api/controllers/socialAuth.js]: @Ramadhan0 we need to pass an ID to the token generator.

Comment (MemunaHaruna) [reply src/api/controllers/socialAuth.js]: @Ramadhan0 Also, why do we need separate variables for `existingUser`, `tempUser` and `loginUser`? There seems to be a lot of repetition in these variables.

File: src/api/models/user.js (hunk 14 above)
Comment (MemunaHaruna) [reply src/api/models/user.js]: @Ramadhan0 the feedback @nkpremices gave above doesn't seem to have been addressed.

File: src/api/routes/authRouter.js [hunk 38]
```diff
@@ -1,9 +1,31 @@
...
+import socialAuthController from '../controllers/socialAuth';
+import env from '../../configs/environments';
+import social from '../../middlewares/social/social';
 
 const authRouter = Router();
 
 authRouter.post('/signup', authController.signup);
 authRouter.post('/login', authController.login);
 
+authRouter.post('/testfacebook', social, socialAuthController.facebookAuth);
```
Comment (MemunaHaruna) [inline src/api/routes/authRouter.js]: what do these different routes do? Let's make their purpose obvious from their names.

File: src/index.js [hunk 39]
```diff
@@ -6,7 +6,6 @@ import register from './middlewares/register.app';
 import { sequelize } from './api/models/index';
 import environnements from './configs/environments';
```
Comment (MemunaHaruna) [inline src/index.js]: @Ramadhan0 please, remove this file from your changes as it's not related to your task.

File: src/middlewares/social/social.js [hunk 40]
```diff
@@ -0,0 +1,39 @@
+/**
+ * damydata for morking social login on Tacebook Twitter and Google
```
Comment (MemunaHaruna) [inline src/middlewares/social/social.js]: @Ramadhan0 `dummy data for mocking social login on Facebook, Twitter and Google`

Please, pay attention to your spellings. Also, can you change the name of this file to better represent its purpose?

File: test/home.test.js [hunk 41]
```diff
@@ -2,15 +2,14 @@ import chai from 'chai';
 import chaiHttp from 'chai-http';
 
 import server from '../src/index';
-import env from '../src/configs/environments';
```
Comment (MemunaHaruna) [inline test/home.test.js]: @Ramadhan0 Not sure why you have changes in this file.

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: src/api/controllers/password.reset.js (hunk 34 above)
Comment (Ramadhan0) [reply src/api/controllers/password.reset.js]: I worked with Premise and we decided to remove and add it in dotenv

File: src/api/controllers/socialAuth.js (hunk 35 above)
Comment (Ramadhan0) [reply src/api/controllers/socialAuth.js:32]: Alain is going to use this profile URL from social media.
Alain proposed that:  " if a user register using social link ie. facebook, the profile picture will automatically be the one from facebook until he or she decided to change the profile picture"

File: src/api/controllers/socialAuth.js (hunk 36 above)
Comment (Ramadhan0) [reply src/api/controllers/socialAuth.js:66]: we need to know where a user registered from ie, facebook twitter or google

File: test/home.test.js (hunk 41 above)
Comment (Ramadhan0) [reply test/home.test.js]: This file was testing Dotenv, that file contain Dotenv configurations.

File: src/index.js (hunk 39 above)
Comment (Ramadhan0) [reply src/index.js]: Premise gave me feedback yesterday about third-party routes I had on index.js while we have a specific file for it ```src/middlewares/register.app.js``` so, I moved those files to register.app.js.

File: package.json [hunk 42]
```diff
@@ -41,16 +41,24 @@
     "@sendgrid/mail": "^6.4.0",
     "babel-plugin-istanbul": "^5.1.2",
     "bcrypt": "^3.0.6",
-    "body-parser": "^1.18.3",
+    "body-parser": "^1.19.0",
```
Comment (MemunaHaruna) [inline package.json]: @Ramadhan0 why have you updated body-parser?

File: .env.example [hunk 43]
```diff
@@ -10,6 +10,21 @@ TOKEN_KEY=TheTokenKey
 MAILER_API_KEY=theTokenOfTheMailerService
 MAILER_EMAIL=theEmailUsedByTheMailerAsSender
 API_BASE_URL=theApiBaseUrlToInsertInTheLinksSharedInEmails
-TOKEN_KEY=TheTokenKey
-
 BCRYPT_HASH_ROUNDS=theBcryptSaltingRounds
+
+
+
+facebookClientId=FACEBOOK_APP_ID
```
Comment (MemunaHaruna) [inline .env.example]: @Ramadhan0 so that we maintain the same format as the other env variables, change these to:

```
FACEBOOK_CLIENT_ID=theFacebookClientId
GOOGLE_CLIENT_SECRET=theGoogleClientSecret
TWITTER_SESSION_KEY=theTwitterSessionKey
```

please, use this format for the rest.

File: .eslintrc.js [hunk 44]
```diff
@@ -1,69 +0,0 @@
-// Find more about the rules below here https://eslint.org/docs/rules/
-module.exports = {
```
Comment (MemunaHaruna) [inline .eslintrc.js]: @Ramadhan0 why is this file deleted? Please, rebase/merge with develop.

File: .hound.yml [hunk 45]
```diff
@@ -1,5 +1,5 @@
 fail_on_violations: true
 eslint:
- enabled: true
- config_file: .eslintrc.json
- ignore_file: .eslintignore
+  enabled: true
```
Comment (MemunaHaruna) [inline .hound.yml:3]: @Ramadhan0 this file shouldn't be in your PR.

File: src/api/controllers/password.reset.js [hunk 46]
```diff
@@ -59,16 +59,14 @@ export default class PasswordReset {
...
-    await User.update(
-      {
-        password: hashedPassword
-      },
-      {
-        where: {
-          email: userEmail
-        }
+    await User.update({
+      password: hashedPassword
```
Comment (MemunaHaruna) [inline src/api/controllers/password.reset.js]: @Ramadhan0 this file also shouldn't be in your PR. Please, apply this feedback to all other files not directly related to your PR.

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: .hound.yml (hunk 45 above)
Comment (Ramadhan0) [reply .hound.yml:3]: Here, me and Premise we were switching between ```.eslintrc.json``` and ```.eslintrc.js``` in order to ensure how those files were working.

Comment (MemunaHaruna) [reply .hound.yml:3]: Alright, let's undo the changes here since it's not related to your PR.

File: src/middlewares/register.app.js [hunk 47]
```diff
@@ -17,6 +23,12 @@ export default (app) => {
     // Parse req object and make data available on req.body
     .use(bodyParser.json())
     .use(bodyParser.urlencoded({ extended: true }))
+    .use(session({
+      // session
+      saveUninitialized: true,
+      secret: process.env.secretOrKey
```
Comment (MemunaHaruna) [inline src/middlewares/register.app.js]: Is `secretOrKey` still the name of the env variable? I think it's now `TWITTER_SESSION_KEY` but please confirm.

File: src/docs/swagger.json [hunk 48]
```diff
@@ -81,6 +81,69 @@
...
+          "201": {
+            "description": "Account Successfully created"
+          },
+          "500": {
+            "description": "Server error"
+          }
+        }
+      }
+    },
+    "/auth/google": {
```
Comment (MemunaHaruna) [inline src/docs/swagger.json]: Do the actual routes have `/auth` in them?

**Review by MemunaHaruna — APPROVED** [review]

File: src/api/controllers/socialAuth.js [hunk 49]
```diff
@@ -0,0 +1,117 @@
...
+    });
+    if (!existingUser) {
+      existingUser = User.create({
+        username: req.user.displayName,
+        email: req.user.emails[0].value,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+      const token = generateToken({ username: req.user.displayName, id: req.user.id },
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js]: here it is nor the id from the req.user but from the table's dataValues

File: src/api/controllers/socialAuth.js [hunk 50]
```diff
@@ -0,0 +1,117 @@
...
+        process.env.TOKEN_KEY);
+      return res.status(status.CREATED).json({ user: { ...existingUser.get(), token } });
+    }
+    const token = generateToken({
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js:77]: why did you repeat the generate token and send the same result twice???

File: src/api/controllers/socialAuth.js [hunk 51]
```diff
@@ -0,0 +1,117 @@
...
+      }
+    });
+    if (!existingUser) {
+      existingUser = User.create({
+        username: req.user.username,
+        profileImage: req.user.photos[0].value,
+        provider: req.user.provider,
+        uniqueId: req.user.id
+      });
+      const token = generateToken({ username: req.user.displayName, id: req.user.id },
```
Comment (alainburindi) [inline src/api/controllers/socialAuth.js]: the same here

File: src/middlewares/social/social.js [hunk 52]
```diff
@@ -0,0 +1,39 @@
+/**
```
[the remark this answers is not in the document]
Comment (alainburindi) [reply src/middlewares/social/social.js:1]: you still have hound issue

**Review by alainburindi — CHANGES_REQUESTED** [review]

**Comment by truestbyheart** [comment]
Please put the ``get`` request in caps ``GET`` in your PR description.

**Comment by truestbyheart** [comment]
Please can you refer to @MCFrank16  or @truestbyheart PR to see how they have written this part.

File: package.json [hunk 53]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "start": "babel-node src/index.js",
+    "setup": "NODE_ENV=test npm run migrate:undo  && NODE_ENV=test npm run migrate && NODE_ENV=test npm run seed",
```
Comment (MemunaHaruna) [inline package.json]: @Ramadhan0 why do you need this?

File: .env.example [hunk 54]
```diff
@@ -10,10 +10,28 @@ TOKEN_KEY=TheTokenKey
 MAILER_API_KEY=theTokenOfTheMailerService
 MAILER_EMAIL=theEmailUsedByTheMailerAsSender
 API_BASE_URL=theApiBaseUrlToInsertInTheLinksSharedInEmails
-TOKEN_KEY=TheTokenKey
```
Comment (MemunaHaruna) [inline .env.example:13]: @Ramadhan0 why have you removed this env variable? I can see it's being used in this file: src/api/controllers/socialAuth.js

File: src/api/migrations/20190620192200-create-user.js [hunk 55]
```diff
@@ -68,5 +67,6 @@ export default {
       type: Sequelize.DATE
     }
   }),
+  // eslint-disable-next-line no-unused-vars
```
Comment (MemunaHaruna) [inline src/api/migrations/20190620192200-create-user.js]: @Ramadhan0 why are you disabling this eslint rule?

File: src/api/migrations/20190620192200-create-user.js [hunk 56]
```diff
@@ -7,8 +7,7 @@ export default {
       type: Sequelize.INTEGER
     },
     username: {
-      type: Sequelize.STRING,
-      unique: true
+      type: Sequelize.STRING
```
Comment (MemunaHaruna) [inline src/api/migrations/20190620192200-create-user.js]: @Ramadhan0 username should be unique.

File: src/api/migrations/20190624064739-create-test-articl.js [hunk 57]
```diff
@@ -0,0 +1,51 @@
+export default {
```
Comment (MemunaHaruna) [inline src/api/migrations/20190624064739-create-test-articl.js]: @Ramadhan0 why do you have this file in your work?

File: src/api/models/test-articl.js [hunk 58]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (MemunaHaruna) [inline src/api/models/test-articl.js]: @Ramadhan0 same with this file. why is it here?

File: .env.example (hunk 54 above)
Comment (Ramadhan0) [reply .env.example:13]: it is there on line 8
